drivers/watchdog/wdt_mchp_xec.c

```diff
@@ -173,9 +191,15 @@ static int wdt_xec_init(const struct device *dev)
 	return 0;
 }
 
+static const struct wdt_xec_config wdt_xec_config_0 = {
+	.regs = (struct wdt_regs *)(DT_INST_REG_ADDR(0)),
+	.girq = DT_INST_PROP_BY_IDX(0, girqs, 0),
+	.girq_pos = DT_INST_PROP_BY_IDX(0, girqs, 1),
+};
+
 static struct wdt_xec_data wdt_xec_dev_data;
 
 DEVICE_DT_INST_DEFINE(0, wdt_xec_init, NULL,
-		    &wdt_xec_dev_data, NULL,
+		    &wdt_xec_dev_data, &wdt_xec_config_0,
```

Comment:
This driver is single instance, please add a build assert that the number of instances is only 1.

Example from busy_sim.c
```
BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(vnd_busy_sim) == 1,
	     "add exactly one vnd,busy-sim node to the devicetree");
```

Reply:
Added the build assert.